test/integration/change-streams/change_stream.test.ts

```diff
@@ -2138,6 +2276,144 @@ describe('ChangeStream resumability', function () {
         );
       });
     });
+
+    context('#asyncIterator', function () {
+      for (const { error, code, message } of resumableErrorCodes) {
+        it(
+          `resumes on error code ${code} (${error})`,
+          { requires: { topology: '!single', mongodb: '>=4.2' } },
+          async function () {
+            changeStream = collection.watch([]);
+            await initIteratorMode(changeStream);
+            const changeStreamIterator = changeStream[Symbol.asyncIterator]();
+
+            await client.db('admin').command({
+              configureFailPoint: is4_2Server(this.configuration.version)
+                ? 'failCommand'
+                : 'failGetMoreAfterCursorCheckout',
+              mode: { times: 1 },
+              data: {
+                failCommands: ['getMore'],
+                errorCode: code,
+                errmsg: message
+              }
+            } as FailPoint);
+
+            await collection.insertOne({ city: 'New York City' });
+            await changeStreamIterator.next();
+
+            expect(aggregateEvents).to.have.lengthOf(2);
+
+            changeStream.close();
```

Comment:
same comment about closing the change stream as above.  this applies to all tests you've added actually